gateway-service-impl/src/main/java/org/hypertrace/gateway/service/explore/RequestHandler.java

```diff
@@ -227,6 +241,62 @@ private List<String> getEntityIdsToFilterFromSourceEDS(
         .collect(Collectors.toUnmodifiableList());
   }
 
+  private int fetchTotal(
+      ExploreRequestContext requestContext, ExploreRequest request, QueryRequest queryRequest) {
+    QueryRequest totalQueryRequest = buildTotalQueryRequest(request, queryRequest);
+    Iterator<ResultSetChunk> totalIterator = executeQuery(requestContext, totalQueryRequest);
+    while (totalIterator.hasNext()) {
+      ResultSetChunk chunk = totalIterator.next();
+      LOG.info("Total received chunk is: {}", chunk);
+      if (chunk.getRowCount() < 1) {
+        break;
+      }
+
+      if (!chunk.hasResultSetMetadata()
+          || chunk.getResultSetMetadata().getColumnMetadataList().size() != 1
+          || !TOTAL_ALIAS_NAME.equals(
+              chunk.getResultSetMetadata().getColumnMetadata(0).getColumnName())) {
+        LOG.warn("Chunk doesn't have result metadata so couldn't process the response.");
+        break;
+      }
+      return chunk.getRow(0).getColumn(0).getInt();
+    }
+    return 0;
+  }
+
+  private QueryRequest buildTotalQueryRequest(ExploreRequest request, QueryRequest queryRequest) {
+    QueryRequest.Builder builder = QueryRequest.newBuilder();
+    List<org.hypertrace.gateway.service.v1.common.Expression> groupBys =
+        ExpressionReader.getAttributeExpressions(request.getGroupByList());
+    // add all group-by columns as distinct count to get total
```

Comment:
Perhaps, I'm missing something here. Simply removing the sorting and pagination and getting the total count is sufficient, right?

Any reason why we want to pick the `group-by` columns? In a query, say `SELECT a, b FROM x WHERE ... GROUP BY a, b`, how does DISTINCT_COUNT(a) + DISTINCT_COUNT(b) give the total number of rows that query would return? Wouldn't the rows having both a and b be double-counted?